src/fileio/datafile.cxx

```diff
@@ -539,6 +541,8 @@ bool Datafile::write(const char *format, ...) const {
   tmp.openw(filename);
   bool ret = tmp.write();
   tmp.close();
+
+  delete[] filename;
```

Comment:
`filename` is always allocated - maybe it would be cleaner not to do so - but that is a different issue. Shouldn't be needed ...